src/app/store/actions/product.actions.ts

```diff
@@ -0,0 +1,75 @@
+import { createAction, props } from '@ngrx/store';
+import { Product } from 'src/app/entities/product';
+
+export const GetProductList = createAction(
+  '[Product] Load Products'
+)
+
+export const GetProductListSuccess = createAction(
+  '[Product] Success in getting the list of products',
+  props<{ products: Product[] }>()
+);
+
+export const GetProductListeError = createAction(
```

Comment:
I'd advise against making error actions for everything. One will suffice, since you only need to show a small message if the action failed.